### DISCUSSION TITLE
How to deploy git branch using Vercel REST API?

### DISCUSSION THREAD

**Discussion by GorvGoyl** [vercel/community#4240] [Help]
### Summary

My Vercel (Next.js) project is connected to Github. How can I trigger deployment via Vercel rest API of a custom branch let's say dev?

The documentation is too poor for me to understand it: https://vercel.com/docs/rest-api/endpoints#create-a-new-deployment

### Steps to Reproduce

 I searched for existing answers that seemed to be working for others but not for me:

```
curl -X POST "https://api.vercel.com/v13/deployments" \
  -H "Authorization: Bearer xxx" \
  -H "Content-Type: application/json" \
  -d '{
      "gitSource": {
        "ref" : "dev",
        "repoId": "579769583",
        "sha": "539a426",
        "type": "github"
      },
      "name": "chatgpt-writer-site"
      }'

```

Output Error:

```
{"error":{"code":"incorrect_git_source_info","message":"The provided GitHub repository does not contain the requested branch or commit reference. Please ensure the repository is not empty."}}%
```

**Comment by amyegan** [comment]
Can you double check that you have the correct repo id, sha, and branch ref? 

Another thing that sometimes causes problems for folks is a missing team id. If the project belongs to a Pro team rather than a personal Hobby account, make sure you include the `teamId` query param.

I just tried creating a new deployment for my test site and `POST /v13/deployments` worked for me when I used a valid repo id, sha, branch name (ref), and team id. Please let me know if you're still unable to deploy after verifying those values 🙏

**Reply by GorvGoyl** [reply]
I'm a pro account user and i tried adding teamID but still facing the error: "The provided GitHub repository does not contain the requested branch or commit reference. Please ensure the repository is not empty."

I'm not sure what I'm doing wrong.
I got the repoId from looking at the github repo page source:

<img width="944" alt="image" src="https://github.com/vercel/community/assets/7106086/006b37d1-afd4-485c-921c-8e242d75e118">

sha:
![screenshot 2023-09-26 at 23 14 08@2x](https://github.com/vercel/community/assets/7106086/b6552867-21e4-40a6-9836-8e652b8a89f4)


```
curl --location 'https://api.vercel.com/v13/deployments?forceNew=0&skipAutoDetectionConfirmation=0&teamId=team_u8Hy4MZvj7Hkdy6dfagTBOKw' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer xxx' \
--header 'Content-Type: application/json' \
--data '{
    "gitSource": {
        "ref": "dev",
        "repoId": "579769583",
        "sha": "539a426",
        "type": "github"
    },
    "name": "chatgpt-writer-site"
}'
```

**Reply by GorvGoyl** [reply]
Hi, I'm still facing the issue and waiting for your response

**Reply by VincentShao32** [reply]
Hi, not sure if you're still facing this issue, but it seems you haven't copied the full sha. Make sure to click "copy full SHA" next to the sha that you see, it should be much longer. For reference, this was the full sha for one of my commits: 1758cad7aee6a445da137da3dfd71b58880f994f
